chain/near.ts

```diff
@@ -0,0 +1,386 @@
+import '../common/eager_offset'
+import { Bytes } from '../common/collections'
+import { BigInt } from '../common/numbers'
+
+// Most types from this namespace are direct mappings or adaptations from:
+// https://github.com/streamingfast/proto-near/blob/develop/sf/near/codec/v1/codec.proto
+export namespace near {
+  export type CryptoHash = Bytes
+
+  export type AccountId = string
+
+  export type BlockHeight = u64
+
+  export type Balance = BigInt
+
+  export type Gas = u64
+
+  export type ShardId = u64
+
+  export type NumBlocks = u64
+
+  export type ProtocolVersion = u32
+
+  export type Payload = u64
+
+  export enum CurveKind {
+    ED25519 = 0,
+    SECP256K1 = 1,
+  }
+
+  export class Signature {
+    constructor(
+      public kind: CurveKind,
+      public bytes: Bytes,
+    ) {}
+  }
+
+  export class PublicKey {
+    constructor(
+      public kind: CurveKind,
+      public bytes: Bytes,
+    ) {}
+  }
+
+  export enum AccessKeyPermissionKind {
+    FUNCTION_CALL = 0,
+    FULL_ACCESS = 1,
+  }
+
+  export class FunctionCallPermission {
+    constructor(
+      public allowance: BigInt,
+      public receiverId: string,
+      public methodNames: Array<string>,
+    ) {}
+  }
+
+  export class FullAccessPermission {}
+
+  export class AccessKeyPermissionValue {
+    constructor(
+      public kind: AccessPermissionKind,
+      public data: Payload,
+    ) {}
+
+    toFunctionCall(): FunctionCallPermission {
+      assert(this.kind == AccessKeyPermissionKind.FUNCTION_CALL, 'AccessKeyPermissionValue is not a \'FunctionCall\'.')
+      return changetype<FunctionCallPermission>(this.data as u32)
+    }
+
+    toFullAccess(): FullAccessPermission {
+      assert(this.kind == AccessKeyPermissionKind.FULL_ACCESS, 'AccessKeyPermissionValue is not a \'FullAccess\'.')
+      return changetype<FullAccessPermission>(this.data as u32)
+    }
+
+    static fromFunctionCall(input: FunctionCallPersmission): AccessKeyPermissionValue {
+      return new AccessKeyPermissionValue(ActionKind.FUNCTION_CALL, changetype<u32>(input))
+    }
+
+    static fromFullAccess(input: FullAccessPersmission): AccessKeyPermissionValue {
+      return new AccessKeyPermissionValue(ActionKind.FULL_ACCESS, changetype<u32>(input))
+    }
+  }
+
+  export class AccessKey {
+    constructor(
+      public nonce: u64,
+      public permission: AccessKeyPermissionValue,
+    ) {}
+  }
+
+  export class DataReceiver {
+    constructor(
+      public dataId: CryptoHash,
+      public receiverId: string,
+    ) {}
+  }
+
+  export enum ActionKind {
+    CREATE_ACCOUNT = 0,
+    DEPLOY_CONTRACT = 1,
+    FUNCTION_CALL = 2,
+    TRANSFER = 3,
+    STAKE = 4,
+    ADD_KEY = 5,
+    DELETE_KEY = 6,
+    DELETE_ACCOUNT = 7,
+  }
+
+  export class CreateAccountAction {}
```

Comment:
I think it's correct, both looking at:

- The outdated published `near-primitives` crate: https://docs.rs/near-primitives/0.1.0-pre.1/near_primitives/transaction/struct.CreateAccountAction.html
- And the current `master` branch of `nearcore/primitives`: https://github.com/near/nearcore/blob/master/core/primitives/src/transaction.rs#L83

It seems correct, that's probably because creating an account doesn't involve any data besides the new `AccountId`.
Reference docs: https://docs.near.org/docs/concepts/account#compared-to-ethereum